### PR TITLE
Add "Auto Scaling Group With No Associated ELB" query for Terraform and Ansible Closes #2720

### LINKED ISSUE #2720: Add "Auto Scaling Group With No Associated ELB" query for Terraform and Ansible
### Platform
Terraform and Ansible

### Provider
AWS

### Description
AWS Auto Scaling Groups must have associated ELBs to ensure high availability and improve application performance. This means the attribute 'LoadBalancerNames' must be defined and not empty

### PR DESCRIPTION
**Pull request by rafaela-soares**
Closes #2720

**Proposed Changes**

- Support "Auto Scaling Group With No Associated ELB" query for Terraform and Ansible (same as "Auto Scaling Group With No Associated ELB" for CloudFormation). It is necessary to check if the attribute load_balancers is not set or is empty

I submit this contribution under Apache-2.0 license.

### CODE REVIEW

File: assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/metadata.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "8e94dced-9bcc-4203-8eb7-7e41202b2505",
+  "queryName": "Auto Scaling Group With No Associated ELB",
+  "severity": "MEDIUM",
+  "category": "Availability",
+  "descriptionText": "AWS Auto Scaling Groups must have associated ELBs to ensure high availability and improve application performance. This means the attribute 'LoadBalancerNames' must be defined and not empty.",
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/metadata.json:6]: The attribute should be 'load_balancers', no?

File: assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+CxPolicy[result] {
+	resource = input.document[i].resource.aws_autoscaling_group[name]
+
+	object.get(resource, "load_balancers", "undefined") == "undefined"
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("aws_autoscaling_group[%s]", [name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("aws_autoscaling_group[%s].load_balancers is set", [name]),
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego:12]: ...is set and not empty

File: assets/queries/ansible/aws/auto_scaling_group_with_no_associated_elb/metadata.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "050f085f-a8db-4072-9010-2cca235cc02f",
+  "queryName": "Auto Scaling Group With No Associated ELB",
+  "severity": "MEDIUM",
+  "category": "Availability",
+  "descriptionText": "AWS Auto Scaling Groups must have associated ELBs to ensure high availability and improve application performance. This means the attribute 'LoadBalancerNames' must be defined and not empty.",
```
Comment (felipe-avelar) [inline assets/queries/ansible/aws/auto_scaling_group_with_no_associated_elb/metadata.json:6]: The attribute should be 'load_balancers', no?

File: assets/queries/ansible/aws/auto_scaling_group_with_no_associated_elb/query.rego [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	resource := task[modules[m]]
+	ansLib.checkState(resource)
+
+	object.get(resource, "load_balancers", "undefined") == "undefined"
+
+	result := {
+		"documentId": id,
+		"searchKey": sprintf("name={{%s}}.{{%s}}", [task.name, modules[m]]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("%s.load_balancers is set", [modules[m]]),
```
Comment (felipe-avelar) [inline assets/queries/ansible/aws/auto_scaling_group_with_no_associated_elb/query.rego:18]: ...is set and not empty

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Check comments, please

File: assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/metadata.json (hunk 1 above)
Comment (rafaela-soares) [reply assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/metadata.json]: Yes, sorry. Done 👍

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**